### PR TITLE
[BACKLOG-21152] Create a new datasource type for the CDE renderer that interacts with Solr

### PR DESCRIPTION
**Pull request by davidmsantos90**
@webdetails/millenniumfalcon please review

### CODE REVIEW

File: pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      return "http://172.20.43.56:8983/solr";
+    },
+
+    /**
+     * Gets the Apache Solr Service Collection
+     *
+     * @return {string} the Solr Collection
+     */
+    getCollection: function() {
+      return "tenis";
```
Comment (dcleao) [inline pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js]: What's the whole point of a default collection? Is there a default collection name in all SOLR installations that could be a good default? And why would that be a per-environment default?

File: pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "use strict";
+
+  return {
+    /**
+     * Gets the Apache Solr Service Endpoint
+     *
+     * @return {string} the Solr Endpoint
+     */
+    getEndpoint: function() {
+      return "http://172.20.43.56:8983/solr";
```
Comment (dcleao) [inline pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js]: I know this is probably the best we can do at the moment, but this deserves a very well explained comment on why, when, how, etc...

File: pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * the license for the specific language governing your rights and limitations.
+ */
+
+define([], function() {
+
+  "use strict";
+
+  return {
+    /**
+     * Gets the Apache Solr Service Endpoint
```
Comment (dcleao) [inline pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js]: Please terminate with a period character.

File: pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @return {string} the Solr Endpoint
```
Comment (dcleao) [inline pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js]: Please terminate with a period character and start with upper case.

File: pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    getEndpoint: function() {
+      return "http://172.20.43.56:8983/solr";
+    },
+
+    /**
+     * Gets the Apache Solr Service Collection
```
Comment (dcleao) [inline pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js]: Idem.

File: pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js (hunk 1 above)
Comment (davidmsantos90) [reply pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js]: I'm going to remove this file, as it doesn't make sense for the solr use case.

File: pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js (hunk 2 above)
Comment (davidmsantos90) [reply pentaho-js/src/main/javascript/cdf/queries/SolrQuery.ext.js]: I'm going to remove this file, as it doesn't make sense for the solr use case.

And we can configure this property in the editor to change its value

**Comment by davidmsantos90** [comment]
@webdetails/millenniumfalcon unit tests completed. please review

File: core-js/src/main/javascript/cdf/dashboard/Dashboard.dataSources.js [hunk 6]
```diff
@@ -87,7 +87,7 @@ define([
      *
      * @param {string|Object}     name        The name of the data source.
      * @param {Object|boolean}    obj         The data source to be added.
-     * @param {boolean|undefined} force=false A flag indicating if any previous data sources with the
+     * @param {boolean|undefined} [force]=false A flag indicating if any previous data sources with the
```
Comment (dcleao) [inline core-js/src/main/javascript/cdf/dashboard/Dashboard.dataSources.js]: `[force=false]`

**Pull request merged**